semgrep/javascript/nodejs/express/nodejs-express-mysql-injection.yml

```diff
@@ -0,0 +1,32 @@
+rules:
+  - id: nodejs-mysql-body-inection 
+    languages:
+      - javascript
+    message: | 
+      Potential SQL Injection identified. Consider using the connection.escape method built in node-mysql
+    patterns:
+      - pattern-inside:  |
+          $MYSQL=require('mysql')
+          ...
+          $CONN = $MYSQL.createConnection(...)
+          ...
+      - pattern:  |
+          $CONN.query($SQL,...)
+      - pattern-either:
+        - pattern-inside: function ... ($REQ, $RES) {...}
+        - pattern-inside: function ... ($REQ, $RES, $NEXT) {...}
+        - pattern-inside: $APP.get(..., function $FUNC($REQ, $RES) {...})
+        - pattern-inside: $APP.post(..., function $FUNC($REQ, $RES) {...})
+        - pattern-inside: $APP.put(..., function $FUNC($REQ, $RES) {...})
+        - pattern-inside: $APP.head(..., function $FUNC($REQ, $RES) {...})
+        - pattern-inside: $APP.delete(..., function $FUNC($REQ, $RES) {...})
+        - pattern-inside: $APP.options(..., function $FUNC($REQ, $RES) {...})
+      - pattern-regex: \+.*?\+
+      - pattern-not-regex: \+.*?escape.*?\+
+      - pattern: $REQ.body.$DATA
+      - metavariable-pattern:
+          metavariable: $SQL
+          patterns:
+              - pattern-regex: .*\b(?i)(SELECT|CREATE|DROP|INSERT|DELETE|UPDATE|ALTER)\b.*
```

Comment:
isn't it enough that that we use "$CONN.query($SQL,...)"? why do we need to validate it starts with an SQL command? Just want to save the performance of executing another regex and it seems it doesn't add much value?